pulsar-broker/src/main/java/org/apache/pulsar/broker/systopic/SystemTopicClient.java

```diff
@@ -93,22 +93,22 @@
          * @return message id
          * @throws PulsarClientException exception while write event cause
          */
-        MessageId write(T t) throws PulsarClientException;
+        MessageId write(T t, String key) throws PulsarClientException;
```

Comment:
+1, why mark this one as resolved, I don't see any conclusion for this one.
And it's better to use `MessageId write(String key, T value)`.

Reply:
Because `MessageId write(T t) throws PulsarClientException;` can not be implement in the `TransactionBufferSnapshotBaseSystemTopicClient`.
We can not get the key by the type T.